### PR TITLE
Mathmagician: Testing

### PR DESCRIPTION
**Pull request by JoyTaribagshaw**
In this branch, I implemented the following;
>

- Add tests to the Math Magicians app using [Jest](https://jestjs.io/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro).

- [ ] Set up React Testing Library following the instructions in the official [documentation](https://testing-library.com/docs/react-testing-library/intro).

- [ ] Write unit tests for the files operate.js and calculate.js using Jest.

- Created unit tests for all React components:

- [ ] Used Jest snapshots to test the components.
- [ ] Used React Testing Library to simulate user interaction.

- Ran my full tests suite using Jest. All tests passed.

Tests

### CODE REVIEW

**Comment by KDlamini** [comment]
## Status: Invalid 🔴 

### Hi @JoyTaribagshaw 🙋‍♀️,  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
 
You have submitted a project, that does not have equal commits from all coding partners. This milestone is a [pair-programming](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#work-and-submission-mode) activity. Kindly ensure that everyone has contributed code to the project in order for this submission to be valid.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// Test the four major math operations (add, subtract, multiply, & divide)
+// Also include modulo and modulo errors
+describe('Test the operate function', () => {
+  // Test the addition operation
+  test('Sum 3 + 7 to equal 10', () => {
+    // Arrange the numbers and operations
+    const numberOne = 3;
+    const numberTwo = 7;
+    const operation = '+';
+    const output = '10';
```
Comment (roshan-bajgain) [inline src/tests/calculate.test.js]: - [ ] [Optional] Please make your test file `operate.test.js` as you are testing from the `operate.js` file but you name the file `calculate.test.js`

File: src/tests/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import calculate from '../logic/calculate';
+
+const calculateObj = {
+  total: 0,
+  next: null,
+  operation: null,
+};
+
```
Comment (roshan-bajgain) [inline src/tests/operate.test.js]: - [ ] [Optional] Please make sure to rename the file as` calculate.test.js` as you are testing the `calculate.js` file but you name it `operate.test.js`

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @JoyTaribagshaw and @hetrox8, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Some test are passing ✔️ 

## Required Changes ♻️
- [ ] If you see the project requirement then you need to Create unit tests for all React components but I did not see test for `Navbar` and `header` please write test for it and you are good to go.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/operate.test.js (hunk 2 above)
Comment (JoyTaribagshaw) [reply src/tests/operate.test.js]: Thank you for the review @roshan-bajgain , I have made the required changes.

File: src/tests/calculate.test.js (hunk 1 above)
Comment (JoyTaribagshaw) [reply src/tests/calculate.test.js]: Thank you for the review @roshan-bajgain , I have made the required changes.

**Review by ademibowale — APPROVED** [review]
Hi @JoyTaribagshaw  , 


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️
- There should be the usage of React Testing Library to simulate user interaction.✔️
- There should be the usage of Jest snapshots for testing all the components. ✔️
- Descriptive commit messages ✔️ 
- Good use of Github flow ✔️






- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**